src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -18,6 +18,7 @@ public partial class StackTrace
     {
         public const int METHODS_TO_SKIP = 0;
 
+        private static readonly bool s_ilOffsetToStackTrace = Environment.GetEnvironmentVariable("COMPlus_ILOffsetToStackTrace") == "1";
```

Comment:
A static constructor was used to avoid unnecessarily checking the environment variable multiple times. Could this be a problem?

Reply:
Yes, this could cause problems when for example Environment .cctor crashes. Using env variables will also require them to be set every time or won't work on deployments where setting env variables are not supported/allowed. I think using https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/LocalAppContextSwitches.cs would be better.

/cc @jkotas

Reply:
The code that gets the Environment value from the static constructor is already used in various runtimes.
see : https://github.com/dotnet/runtime/blob/1d12854807aaedbd0019b34149c3633d462f4ca5/src/libraries/System.Private.CoreLib/src/System/Diagnostics/DebugProvider.Unix.cs#L8

Also, if there is a crash in Environment .cctor, it is thought that there is a problem in the system already. Should this be considered?

Reply:
@marek-safar All review contents have been applied except for the use of static constructor. If it is necessary to change the static constructor to LocalAppContextSwitches, I will apply that as well. Please give me your opinions.

Reply:
StackTrace is used (and usable) for crashes in the Environment as well so creating a circular dependency between these types is not desireable.

Reply:
OK, I'll change it to use LocalAppContextCaches.